content/blog/2022/shared-elements.md

```diff
@@ -0,0 +1,420 @@
+---
+title: Every Transition is a Page Transition?
+sub: Experimenting with the shared element transitions API
+author: miriam
+date: 2022-06-27
+tags:
+  - Article
+  - CSS
+  - CSSWG
+  - JavaScript
+layers:
+  caption: |
+    Layers can be written in any order,
+    with the same layer being used in multiple locations,
+    but CSS will treat layers as groups
+    based on the established layer-order --
+    which we can visualize sorted into groups
+    such that the 'last-takes-precedence'.
+  gallery:
+    - img: blog/2022/element-transitions/original.png
+      alt: |
+        Sorting Cascade Layers, original order,
+        '@layer reset, components, utilities;'
+        followed by
+        9 color-coded @layer blocks and imports
+        in random order
+    - img: blog/2022/element-transitions/sorted.png
+      alt: |
+        Sorting Cascade Layers,
+        sorted so last-takes-precedence,
+        '@layer reset, components, utilities;'
+        followed by
+        9 color-coded @layer blocks and imports
+        grouped together by layer,
+        with unlayered styles at the end
+summary: |
+  There's a new web API proposal
+  for transitioning shared-elements
+  across pages.
+  It's great for making smooth page transitions,
+  but what if we apply it to
+  individual elements
+  with changing styles on a single page?
+---
+
+{% import 'embed.macros.njk' as embed %}
+
+Over the last couple years,
+the [Web Incubator Community Group](https://github.com/WICG)
+has been developing a proposal for
+[Shared Element Transitions](https://github.com/WICG/shared-element-transitions).
+The stated goal
+is to smooth-out transitions
+across page-loads on the web --
+helping users maintain context as they navigate.
+From [the explainer](https://github.com/WICG/shared-element-transitions/blob/main/explainer.md):
+
+> When a user navigates on the web from Page-A to Page-B,
+> the viewport jumps and there is a flash of white
+> as elements disappear only to reappear in the same place
+> in some in-progress state.
+> This sequenced, disconnected user experience is disorienting…
+
+<video src="https://user-images.githubusercontent.com/93594/141100217-ba1fa157-cd79-4a9d-b3b4-67484d3c7dbf.mp4" controls="controls" muted="muted">
+</video>
```

Comment:
Interesting that this is the first time we're using `<video>` directly?

Reply:
Yeah, I was surprised as well. I hope there wasn't a good reason for that? I could upload to vimeo, if we think there's a good reason to.